general-concepts/ebuild-revisions/text.xml

```diff
@@ -8,37 +8,98 @@
-time; developers should use their best judgement in these circumstances.
+The developers are encouraged to use common sense when determining
+whether to introduce a new <c>-rX</c> revision. The following rule
+of thumb could be used as a guideline:
 </p>
+<ol>
+  <li>
+    If the change is likely to break the package and require users to
+    downgrade (especially if the package is stable, all non-trivial
+    changes are considered as such), then a new revision should be
+    introduced and the old one kept. If the package has stable keywords,
+    the new revision should be dropped to <c>~arch</c>
+    (see <uri link="::keywording#Keywording on Upgrades"/>).
+  </li>
 
-<important>
-For ebuilds marked stable on at least one arch, only trivial edits can be made
-without a bump (e.g. typo fixes in elog messages). Even simple changes may
-result in a breakage. <b>Modifying stable ebuilds should be avoided.</b>
-</important>
+  <li>
+    If the change makes a substantial difference to the user who already
+    installed the package (fixes runtime issues, changes installed files,
+    etc.) and it would not be propagated using other means, then
+    the ebuild should be renamed to a new revision. If the package has
+    stable keywords, they should be kept intact.
+  </li>
 
-<p>
-When doing a revision bump, the usual rules about dropping to <c>~arch</c> apply.
-See <uri link="::keywording#Keywording on Upgrades"/>.
-</p>
+  <li>
+    Otherwise, the change can be done in place in the current revision
+    of the ebuild.
+  </li>
+</ol>
+
+<p>The examples of changes that warrant a new revision are:</p>
+<ul>
+  <li>adding a patch to fix a runtime issue,</li>
+  <li>installing additional files that could be useful to the user,</li>
+  <li>adding a missing runtime dependency to one of the existing blocks,</li>
+  <li>
+    adding a missing build-time dependency that caused the package
+    to be built successfully but incorrectly (e.g. miss some feature),
+  </li>
+  <li>
+    restricting a runtime dependency version, unless the <c>:=</c>
+    subslot operator is going to trigger a rebuild.
+  </li>
+</ul>
+
+<p>The examples of changes that can be done without a revison bump are:</p>
+<ul>
+  <li>
+    adding a patch to fix a build-time issue that prevented users from
+    building the package (since it does not affect users who already
+    managed to build it),
+  </li>
+  <li>adding a trivial documentation fix,</li>
+  <li>
+    installing an addition file of relatively little value (minor
+    documentation, editor syntax file, bash completion) compared
+    to the cost of rebuilding the package (especially if a new bump
+    is expected soon),
+  </li>
+  <li>
+    adding a missing build-time dependency that caused the package to
+    fail to build,
+  </li>
+  <li>
+    adding a new USE flag or removing an existing one (since change
+    in USE flags is going to trigger <c>--changed-use</c> rebuild),
```

Comment:
If you're going to add a flag whose default setting maintains the status quo, you can just wait until the next version or the next time you need to do a revision bump for some other reason before you mess with the flag. That can be used to avoid rebuilds in extreme cases like libreoffice and chromium. Most other packages, it doesn't really matter what you do, because quick rebuilds don't bother anyone.